src/index.js

```diff
@@ -25,9 +19,17 @@ class Ipid {
             const { path } = await this.#ipfs.name.resolve(identifier, {});
             const [{ content }] = await this.#ipfs.get(path, {});
 
-            return JSON.parse(content.toString());
-        } catch (e) {
-            throw new InvalidDid(did);
+            const parsedContent = JSON.parse(content.toString());
+
+            assertDocument(parsedContent);
+
+            return parsedContent;
+        } catch (err) {
+            if (err.code === 'INVALID_DOCUMENT') {
+                throw err;
+            }
+
+            throw new InvalidDid(did, `Unable to resolve document with did: ${did}`);
```

Comment:
I think it would be nice to forward the original error as `originalError` prop in the error object.